src/components/v2v3/V2V3Project/ProjectDashboard/hooks/usePayProjectModal/usePrepareDelegatePayMetadata.ts

```diff
@@ -0,0 +1,117 @@
+import { useUniswapPriceQuery } from 'components/AMMPrices/hooks/useERC20UniswapPrice'
+import { BUYBACK_DELEGATE_ENABLED_PROJECT_IDS } from 'constants/buybackDelegateEnabledProjectIds'
+import { DEFAULT_ALLOW_OVERSPENDING } from 'constants/transactionDefaults'
+import { JB721DelegateContractsContext } from 'contexts/NftRewards/JB721DelegateContracts/JB721DelegateContractsContext'
+import { ProjectMetadataContext } from 'contexts/shared/ProjectMetadataContext'
+import { BigNumber } from 'ethers'
+import { useContext } from 'react'
+import { encodeDelegatePayMetadata } from 'utils/delegateMetadata/encodeDelegateMetadata'
+import { parseWad } from 'utils/format/formatNumber'
+import { MAX_RESERVED_RATE } from 'utils/v2v3/math'
+import { ProjectCartNftReward } from '../../components/ProjectCartProvider'
+import { useProjectContext } from '../useProjectContext'
+
+export function usePreparePayDelegateMetadata(
+  weiAmount: BigNumber,
+  {
+    nftRewards,
+    receivedTickets,
+  }: {
+    nftRewards: ProjectCartNftReward[]
+    receivedTickets: string | undefined
+  },
+) {
+  const { tokenAddress, tokenSymbol, fundingCycleMetadata } =
+    useProjectContext()
+  const { projectId } = useContext(ProjectMetadataContext)
+
+  const { data: priceQuery } = useUniswapPriceQuery({
+    tokenAddress,
+    tokenSymbol,
+  })
+
+  const { version: JB721DelegateVersion } = useContext(
+    JB721DelegateContractsContext,
+  )
+  const tierIdsToMint = nftRewards
+    .map(({ id, quantity }) => Array<number>(quantity).fill(id))
+    .flat()
+
+  // Total tokens that should be minted by pay() tx, including reserved tokens
+  const requiredTokens = fundingCycleMetadata
```

Comment:
everything in here should be in a usememo imo

Reply:
same here, its just a fixed number of multiplication operations, `O(1)` - the cost of spawning a useMemo might be more than the computation itself 😅